### PR TITLE
Update Compiled Autograd tutorial with images

### LINKED ISSUE #3107: Compiled autograd tutorial review feedback
### 🚀 Describe the improvement or the new tutorial

Filing an issue to track feedback raised in the 2.5 Release Blog review. Error logs as strings aren't easy to read.

### Existing tutorials on this topic

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by xmfan**
Fixes #3107 

## Description
I'm replacing the long string errors with annotated images, with some minor additions:
e.g.

Before
image

After
image

## Checklist

- [X] The issue that is being fixed is referred in the description (see above "Fixes #ISSUE_NUMBER")
- [X] Only one issue is addressed in this pull request
- [X] Labels from the issue that this PR is fixing are added to this pull request
- [X] No unnecessary issues are included into this pull request.

### CODE REVIEW

**Review by albanD — APPROVED** [review]
That is amazing! Thanks for taking the time to do these improvements!

File: intermediate_source/compiled_autograd_tutorial.rst [hunk 1]
```diff
@@ -181,7 +129,7 @@ Or you can use the context manager, which will apply to all autograd calls withi
 
 Compiled Autograd addresses certain limitations of AOTAutograd
 --------------------------------------------------------------
-1. Graph breaks in the forward pass lead to graph breaks in the backward pass:
+1. Graph breaks in the forward pass no longer always lead to graph breaks in the backward pass:
```
Comment (svekars) [inline intermediate_source/compiled_autograd_tutorial.rst:132]:
```suggestion
1. Graph breaks in the forward pass no longer necessarily lead to graph breaks in the backward pass:
```

File: intermediate_source/compiled_autograd_tutorial.rst [hunk 2]
```diff
@@ -97,62 +97,10 @@ Run the script with the ``TORCH_LOGS`` environment variables:
 Rerun the snippet above, the compiled autograd graph should now be logged to ``stderr``. Certain graph nodes will have names that are prefixed by ``aot0_``,
 these correspond to the nodes previously compiled ahead of time in AOTAutograd backward graph 0, for example, ``aot0_view_2`` corresponds to ``view_2`` of the AOT backward graph with id=0.
 
+In the image below, the red box encapsulates the AOT backward graph that is captured by torch.compile without Compiled Autograd.
```
Comment (svekars) [inline intermediate_source/compiled_autograd_tutorial.rst:100]:
```suggestion
In the image below, the red box encapsulates the AOT backward graph that is captured by ``torch.compile`` without Compiled Autograd.
```

**Review by svekars — COMMENTED** [review]
Minor nits, otherwise looks good - let's wait for the preview to check.

**Comment by xmfan** [comment]
Seems to have rendered correctly: https://docs-preview.pytorch.org/pytorch/tutorials/3108/intermediate/compiled_autograd_tutorial.htm

I'll apply the suggestions

**Review by svekars — APPROVED** [review]

**Pull request merged**